x/ccv/provider/keeper/provider_consensus.go

```diff
@@ -0,0 +1,56 @@
+package keeper
+
+import (
+	"cosmossdk.io/math"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/cosmos/interchain-security/v5/x/ccv/provider/types"
+)
+
+// SetLastProviderConsensusValidator sets the given validator to be stored
+// as part of the last provider consensus validator set
+func (k Keeper) SetLastProviderConsensusValidator(
+	ctx sdk.Context,
+	validator types.ConsumerValidator,
+) {
+	k.setValidator(ctx, []byte{types.LastProviderConsensusValsPrefix}, validator)
+}
+
+// SetLastProviderConsensusValSet resets the stored last validator set sent to the consensus engine on the provider
+// to the provided nextValidators.
```

Comment:
nit:
```suggestion
// to the provided `nextValidators`.
```

similar to how you have `providerConsAddr` later on.